### ISSUE TITLE
DotNetObjectReference.Create should disallow null value

### ISSUE THREAD

**Issue by arivoir** [aspnet/AspNetCore#13797]
I have a component that is exposing a _JSInvokable_ method. When calling it from JavaScript it throws the following exception.

> blazor.server.js:8 Uncaught (in promise) Error: System.ArgumentException: Cannot be null, empty, or whitespace. (Parameter 'AssemblyName')
>    at Microsoft.JSInterop.DotNetDispatcher.GetCachedMethodInfo(AssemblyKey assemblyKey, String methodIdentifier)
>    at Microsoft.JSInterop.DotNetDispatcher.InvokeSynchronously(String assemblyName, String methodIdentifier, Object targetInstance, String argsJson)
>    at Microsoft.JSInterop.DotNetDispatcher.BeginInvoke(String callId, String assemblyName, String methodIdentifier, Int64 dotNetObjectId, String argsJson)
>     at Object.endInvokeDotNetFromJS (blazor.server.js:8)
>     at e.<anonymous> (blazor.server.js:8)
>     at blazor.server.js:1
>     at Array.forEach (<anonymous>)
>     at e.invokeClientMethod (blazor.server.js:1)
>     at e.processIncomingData (blazor.server.js:1)
>     at e.connection.onreceive (blazor.server.js:1)
>     at WebSocket.i.onmessage (blazor.server.js:1)

In other libraries this same is working. Maybe Blazor  is mixing library names. Because the "AssemblyName" is not specified anywhere I assume this is a Blazor bug.

**Comment by pranavkm** [comment]
Could you share a minimal sample that reproduces the problem?

**Comment by arivoir** [comment]
This is a strange case as it is working in some cases, can you give me some idea why that exception  could happen? Under what circumstances the 'AssemblyName' can be not specified? Maybe I can create a sample based on that.

**Comment by pranavkm** [comment]
> can you give me some idea why that exception could happen?

The exception message is thrown if you call a static JSInterop method, but don't specify the assembly name:

```js
DotNet.invokeMethodAsync('BlazorSample', 'ReturnArrayAsync')
```

The first argument needs to be null for the exception to be thrown. You could try enabling debug logging to see which one of the JSInterop calls is problematic.

**Comment by stsrki** [comment]
@arivoir Make sure `JSInvokable` method is not inside of a generic class as that is not supported.

**Comment by arivoir** [comment]
I resolved the mystery.
I'm calling an instance method, not a static one. Actually I do not specify the 'AssemblyName' anywhere. Blazor resolves it. 
Because this is an instance method calling. I previously passed the instance of the component using 
`DotNetObjectRef.Create(ComponentReference)`

I had forgotten to initialize the "ComponentReference" property, but the "Create" method never told me about it. It continued and ended up raising the exception I pasted in the original post.

I think it worth adding some null checking in the _Create_ method of _DotNetObjectRef_.
